homeassistant/components/switchbot/switch.py

```diff
@@ -85,9 +176,28 @@ def unique_id(self) -> str:
     @property
     def name(self) -> str:
         """Return the name of the switch."""
-        return self._name
+        return self.switchbot_name
 
     @property
-    def extra_state_attributes(self) -> dict[str, Any]:
+    def device_state_attributes(self) -> dict:
         """Return the state attributes."""
-        return {"last_run_success": self._last_run_success}
+        return {
+            "last_run_success": self._last_run_success,
+            "MAC": self._mac,
+            "switch_mode": self.coordinator.data[self._idx]["data"]["switchMode"],
+        }
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device_info of the device."""
+        return {
+            "identifiers": {(DOMAIN, self._mac.replace(":", ""))},
```

Comment:
Use `connections` instead of `identifiers`


from homeassistant.helpers import device_registry as dr


```suggestion
            "connections": {(dr.CONNECTION_NETWORK_MAC, self._mac)},
```